client/components/settings/local_settings_modal_controller.js

```diff
@@ -0,0 +1,26 @@
+class LocalSettingsModalController {
+  constructor (localSettings) {
+    Object.assign(this, { localSettings })
+    this.settings = {}
+  }
+
+  $onInit () {
+    Object.assign(this.settings, this.localSettings.getFromLocalStorage())
+  }
+
+  loadSettings () {
+    this.settings = this.localSettings.getFromLocalStorage()
+    this.settingsData = Object.keys(this.settings).map(namespace => {
```

Comment:
Why is this object needed? Why don't you use `ngRepeat` on `this.settings`? See [here](https://stackoverflow.com/questions/15127834/how-to-iterate-over-the-keys-and-values-with-ng-repeat-in-angularjs)